### PR TITLE
GH-41201: [C++] Fix mistake in integration test. Explicitly cast std::string to avoid compiler interpreting char* -> bool

### PR DESCRIPTION
**Pull request by lidavidm**
* GitHub Issue: #41201

### CODE REVIEW

File: cpp/src/arrow/flight/integration_tests/test_integration.cc [hunk 1]
```diff
@@ -878,16 +879,16 @@ class SessionOptionsScenario : public Scenario {
...
+                       {}, SetSessionOptionsRequest{
+                               {{"foolong", std::monostate{}},
+                                {"big_ol_string_list",
+                                 std::string("a,b,sea,dee, ,  ,geee,(づ｡◕‿‿◕｡)づ")}}}));
     ARROW_ASSIGN_OR_RAISE(auto res4, client.GetSessionOptions({}, {}));
     if (res4.session_options !=
         std::map<std::string, SessionOptionValue>{
             {"bardouble", 456.0},
-            {"big_ol_string_list", "a,b,sea,dee, ,  ,geee,(づ｡◕‿‿◕｡)づ"}}) {
+            {"big_ol_string_list", std::string("a,b,sea,dee, ,  ,geee,(づ｡◕‿‿◕｡)づ")}}) {
```
Comment (pitrou) [inline cpp/src/arrow/flight/integration_tests/test_integration.cc:891]: You're probably missing the "key_with_invalid_value" in line 856 above?

**Comment by lidavidm** [comment]
@github-actions crossbow submit verify-rc-source-integration-linux-*

**Comment by lidavidm** [comment]
Looks like this fixes things, minus some flakiness with NodeJS that should be unrelated.

**Review by kou — APPROVED** [review]
+1

(`"..."` was processed as `std::vector<std::string>>` not `std::string` implicitly on these environments, right?)

**Comment by lidavidm** [comment]
> +1
> 
> (`"..."` was processed as `std::vector<std::string>>` not `std::string` implicitly on these environments, right?)

It was implicitly treated as `bool` because a string literal is `char[...]` and the compiler apparently prefers to convert to `char* -> bool` rather than `std::string`

**Comment by kou** [comment]
Thanks! I see!

**Review by raulcd — APPROVED** [review]
The CI jobs are successful and the change looks good to me. Thanks @lidavidm for taking care of this one!

**Pull request merged**